frontends/concrete-python/concrete/fhe/mlir/processors/assign_bit_widths.py

```diff
@@ -88,13 +88,38 @@ def max_encrypted_bitwidth_node(node: Node):
         return normal_p + 1
 
     if name in MULTIPLY and all(value.is_encrypted for value in node.inputs):
-        return normal_p + 1
+        # For operations that use multiply, an additional bit
+        # needs to be added to the bitwidths of the inputs.
+        # For single precision circuits the max of the input / output
+        # precisions will be taken in required_encrypted_bitwidth. For
+        # multi-precision, the circuit partitions will handle the
+        # input and output precisions separately.
+        all_inp_bitwidths = []
+        # Need a loop here to allow typechecking and make mypy happy
+        for inp in node.inputs:
+            dtype_inp = inp.dtype
+            assert isinstance(dtype_inp, Integer)
+            all_inp_bitwidths.append(dtype_inp.bit_width)
+
+        normal_p = max(all_inp_bitwidths)
+
+        # FIXME: This probably does not work well with multi-precision!
```

Comment:
is this ok then ?

Reply:
I don't know if the CP tests are ran with multi precision enabled..

Reply:
(entire test suite is ran twice, one with single precision, one with multi precision)

Reply:
thanks @umut-sahin , then the results are correct with multi-precision, the only issues is that Mul does not break the graph into partitions which can have different precisions (though it could). This should be looked into for #410

Reply:
Yeap, that would be a nice improvement indeed! Another improvement would be to not increase the bit-width by 1 if inputs are already low precision as that bit-width increase is for initial addition/subtraction. So if we do 3b * 3b -> 6b, we can use eint<6> instead of eint<7>, right :thinking:

Reply:
we already do that in the code related to this comment: we take `max(outputprecision, max(input_precisions) + 1)` :smile:

Reply:
Yes what I mean is multiply is not multi precision. Of course it works with the flag but the operator itself remain single preicision.

Reply:
So You can remove that comment.

Reply:
but it will be multi precision at some point no? Then the FIXME would be addressed then